cmd/cli/plugin/feature/activate.go

```diff
@@ -24,22 +30,89 @@ var FeatureActivateCmd = &cobra.Command{
+	FeatureActivateCmd.Flags().BoolVarP(&userAllowsVoidingWarranty, "--permanentlyVoidAllSupportGuaranteesForThisEnvironment", "v", false, "Allow for the permanent voiding of all support guarantees for this environment. For some features, e.g. experimental features, if a user sets the activation status to one that does not match the default activation, all support guarantees for this environment will be permanently voided.")
 }
 
 func featureActivate(cmd *cobra.Command, args []string) error {
 	featureName := args[0]
-	featureGateClient, err := featuregateclient.NewFeatureGateClient()
+
+	fgClient, err := featuregateclient.New()
 	if err != nil {
-		return fmt.Errorf("couldn't get featureGateRunner: %w", err)
+		return fmt.Errorf("get FeatureGate client: %w", err)
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
 	defer cancel()
 
-	if err := featureGateClient.ActivateFeature(ctx, featureName, featuregate); err != nil {
-		return fmt.Errorf("couldn't activate feature %s: %w", featureName, err)
+	feature, err := fgClient.GetFeature(ctx, featureName)
+	if err != nil {
+		return fmt.Errorf("get Feature %s: %w", featureName, err)
 	}
-	cmd.Printf("Feature %s Activated", featureName)
+
+	displayActivationWarnings(feature)
+
+	gates, err := fgClient.GetFeatureGateList(ctx)
+	_, featRef := featuregateclient.FeatureRefFromGateList(gates, featureName)
+
+	var userAllows *bool
+	if cmd.Flag("userAllowsVoidingWarranty").Changed {
+		userAllows = &userAllowsVoidingWarranty
+	}
+	warrantyVoidAllowed := isVoidingWarrantyAllowed(featRef, feature, userAllows)
+
+	err = fgClient.ActivateFeature(ctx, featureName, warrantyVoidAllowed)
+	if err != nil {
+		return fmt.Errorf("activate feature %s: %w", featureName, err)
+	}
+
+	cmd.Printf("Feature %s activated", featureName)
 	return nil
 }
+
+// activationWarnings warns the user that technical preview features are
+// unstable and lack support.
+func displayActivationWarnings(feature *core.Feature) {
+	if feature.Spec.Stability == core.TechnicalPreview {
+		fmt.Println("Warning: Technical preview features are not yet considered stable and therefore do not come with support. Use at your own risk.")
+	}
+}
+
+// isVoidingWarrantyAllowed checks if a warranty will be voided and if so, if it
+// is allowed by the user. Request for permission to void warranty only happens
+// if toggling the Feature activation will void the warranty. Otherwise, it is
+// considered not allowed.
+func isVoidingWarrantyAllowed(ref core.FeatureReference, feature *core.Feature, userAllowsByFlag *bool) bool {
+	stability := feature.Spec.Stability
+	policy := core.GetPolicyForStabilityLevel(stability)
+
+	// Check first if warranty will be voided if Feature is activated.
+	if policy.DefaultActivation == feature.Status.Activated && !ref.PermanentlyVoidAllSupportGuarantees && policy.VoidsWarranty {
```

Comment:
I agree with your intention, but not the suggested change in code. 

The `featureRef.Activate` value is the value before the toggle was made. If it matches the default before user toggles activation, that means the activation will deviate from default. So, when it's equal, it means the change will be away from default.

Reply:
void support warranty happens when activating a feature that is unstable, and the unstable features are always deactivated by default, so we just need to check the intent ie if the feature that voids warranty support is being activated or not

Reply:
Thanks for explaining the logic here. Here's what we decided to do:
```
if policy.VoidsWarranty && !ref.PermanentlyVoidAllSupportGuarantees && !policy.DefaultActivation {
<warranty will be voided, check user permission>
```

Reply:
I thought about this some more, and I have yet another solution. I left a very long comment to explain why. Here's what I ended up with: 

```
// willWarrantyBeVoided checks that activating the Feature will cause warranty to be voided.
// Warranty will be voided only if all the following conditions are met:
//   - The stability policy's VoidsWarranty field value is true. This means changing the activation
//     set point away from default activation (in this case, default activate will have to be false)
//     will void the warranty.
//   - The warranty is not already void. If it's void, then it cannot be voided as it's already void.
//   - The Feature activation set point in the FeatureGate reference is false. If the opposite is
//     true--that the Feature is already set to be activated--then there is nothing to change and
//     the warranty will not be voided.
//   - Activating the Feature will set it different than its default activation setting. This means the
//     default activation of a Feature must be false for the warranty to be voided. On the other hand,
//     if the Feature set point is false, and the default activation is true, then the request to activate
//     the Feature will not void warranty.
//
// If all of the above are true, then the warranty will be voided and the function returns true.
// Otherwise, if any are false, the warranty will not be voided and the function returns false.
func willWarrantyBeVoided(ref core.FeatureReference, feature *core.Feature) bool {
	stability := feature.Spec.Stability
	policy := core.GetPolicyForStabilityLevel(stability)
	return policy.VoidsWarranty && !ref.PermanentlyVoidAllSupportGuarantees && !ref.Activate && !policy.DefaultActivation
}
```

Reply:
is this `policy.VoidsWarranty && !ref.PermanentlyVoidAllSupportGuarantees && !ref.Activate && !policy.DefaultActivation` equivalent to ` policy.VoidsWarranty && !ref.PermanentlyVoidAllSupportGuarantees && policy.DefaultActivation != featureRef.Activate`

Reply:
It's so complicated, I can't tell. Are both correct? If so, I think mine is easier to understand and if you don't understand, I have a very long comment to explain each part.

Reply:
i think this logic can be made more concise, but we can revisit this later